### PR TITLE
Avoid duplicate suffix

### PR DESCRIPTION
**Pull request by cv3d**
```python
from mujincontrollerclient import uriutils
uriutils.GetURIFromPartType("abc.mujin.dae", suffix=".mujin.dae")
```

Current result: `mujin:/abc.mujin.dae.mujin.dae`
Expected result: `mujin:/abc.mujin.dae`

### CODE REVIEW

File: python/mujincontrollerclient/uriutils.py [hunk 1]
```diff
@@ -443,6 +443,9 @@ def __init__(self, **kwargs):
         # guess suffix based on primary key
         if not self._suffix and self._primaryKey.endswith(b'.mujin.dae'):
             self._suffix = u'.mujin.dae'
+        elif self._suffix and self._primaryKey.endswith(self._suffix):
+            log.warn('suffix specified in both the primary key and as a parameter')
+            self._primaryKey = self._primaryKey[:-len(self._suffix)]
```
Comment (bucketzxm) [inline python/mujincontrollerclient/uriutils.py]: This seems not correct. Why do you remove suffix from primary key?  @cv3d

Comment (cv3d) [reply python/mujincontrollerclient/uriutils.py]: @bucketzxm You are right. I fixed that, and I also provided an example to reproduce the bug.

**Comment by bucketzxm** [comment]
> ```python
> from mujincontrollerclient import uriutils
> uriutils.GetURIFromPartType("abc.mujin.dae", suffix=".mujin.dae")
> ```
> 
> Current result: `mujin:/abc.mujin.dae.mujin.dae`
> Expected result: `mujin:/abc.mujin.dae`

The function is to get uri from parttype. If user input the parttype as "abc.mujin.dae", I think we should allow it.
If you think "abc.mujin.dae.mujin.dae" is not what you want, I think you should fix the caller.

**Comment by bucketzxm** [comment]
Another example, we may have ".mujin.json",  so the name "abc.mujin.dae.mujin.json" is totally valid. (Although it's a bad name)

**Comment by ziyan** [comment]
@cv3d I think this change will end up hiding the real bug somewhere else in the system. Someone added `.mujin.dae` to a `partType`, and now it's hidden and harder to discover.

**Comment by ziyan** [comment]
I do, however, thinks that `uriutils.py` needs to be drastically improved @bucketzxm

**Pull request closed without merging**